### DISCUSSION TITLE
Exporting a Chain with Governance Proposal Vote Data

### DISCUSSION THREAD

**Discussion by martinbha** [cosmos/community#59] [Chain Builders (Cosmos SDK)] [answered]
Hello, 
When I try to export a chain `export --home chain-data/validator > gov_export_test.json` after a governance proposal has been successfully submitted, voted on, and passed, there doesn't seem to be any information in the votes section. It always comes up like this `"votes": [],`. 

However, when I try kill the chain and export it before the proposal has passed, I am able to see the list of votes.
```
"votes": [
                {
                    "option": "VOTE_OPTION_YES",
                    "options": [
                        {
                            "option": "VOTE_OPTION_YES",
                            "weight": "1.000000000000000000"
                        }
                    ],
                    "proposal_id": "1",
                    "voter": "cosmos1y95nfdpzj8vdh3dpc4sglnprny3faj0y56jyy6"
                },
                {
                    "option": "VOTE_OPTION_YES",
                    "options": [
                        {
                            "option": "VOTE_OPTION_YES",
                            "weight": "1.000000000000000000"
                        }
                    ],
                    "proposal_id": "1",
                    "voter": "cosmos1fegapd4jc3ejqeg0eu3jk4hvr74hg66076gyyd"
                }
            ]
```

Any help on this matter would be greatly appreciated.

**Comment by facundomedica** [comment, marked as answer]
Hello, the `Tally` function deletes all the votes when the voting period ends

**Reply by martinbha** [reply]
Ok, thank you for that information. 
If I may then ask how are you supposed to keep track of the votes on Governance Proposals then if all of the votes have been deleted after the voting period?
For example https://www.mintscan.io/cosmos/proposals/845 here it lists all of the votes and who voted even after the voting period has ended.

**Reply by facundomedica** [reply]
The chain does not have a need for keeping all those votes in state, there is no use for them after they've been counted. If you need them for something else I think you should store that data off-chain. But if you want to query it directly from the chain then you should query an archive node.

What's your use case?

**Reply by martinbha** [reply]
Thank you for that. There isn't a specific use case, was just curious on how to keep the governance voting data on chain. Thank you for your help though.
